samples/sample_multi_transcode/src/sample_multi_transcode.cpp

```diff
@@ -358,36 +359,32 @@ void Launcher::DoTranscoding()
 {
     mfxStatus sts = MFX_ERR_NONE;
 
-    // get parallel sessions parameters
-    mfxU32 totalSessions = (mfxU32)m_pSessionArray.size();
-    MSDKThread* pthread = 0;
-
-    for (mfxU32 i = 0; i < totalSessions; i++)
+    // Run transcoding threads
+    std::vector<std::future<mfxU32>> thrHandles;
+    thrHandles.reserve(m_pSessionArray.size());
+    for (auto session : m_pSessionArray)
     {
-        pthread = new MSDKThread(sts, TranscodeRoutine, (void *)m_pSessionArray[i]);
-        m_HDLArray.push_back(pthread);
-    }
-
-    // Need to determine overlay threads count first
-    size_t nOverlayThreads = 0;
-    for (size_t i = 0; i < m_pSessionArray.size(); i++)
-    {
-        if (m_pSessionArray[i]->pPipeline->IsOverlayUsed())
-            nOverlayThreads++;
+        thrHandles.emplace_back(std::async(std::launch::async,
+                                TranscodeRoutine, (void *)session));
     }
 
     // Transcoding threads waiting cycle
+    size_t totalSessions = m_pSessionArray.size();
```

Comment:
I don't think we need another alias for size of vector, it just overcomplicates the loop below. I suggest to use ```m_pSessionArray.size()``` instead

Reply:
Done